Intersect.Client/Entities/Player.cs

```diff
@@ -2378,12 +2378,7 @@ protected override bool ShouldDrawHpBar
         {
             get
             {
-                if (ShouldNotDrawHpBar)
-                {
-                    return false;
-                }
-
-                if (IsHovered)
+                if (!ShouldNotDrawHpBar || IsHovered)
```

Comment:
This is 100% not a bug fix, it creates a bug. The `ShouldNotDrawHpBar` logic is not invertible like this.

Reply:
this was previously working and i changed it to according to the PR's review, my fault on this since i didn't fully tested the possible outcomes of changing it in the way that was previously suggested. Currently, the player boolean override `ShouldDrawHpBar` should initially follow the same pattern as it's entity virtual boolean `ShouldDrawHpBar` in order to have the desired outcome

![imagen](https://user-images.githubusercontent.com/17498701/210152987-42133d5e-3698-41ff-ad17-66b1591d904b.png)

if we return false within this method, we won't be able to show up the hp bars when players are damaged nor hovered, since we are returning false before the loop can even check after the rest of conditions ahead during the itineration.

Reply:
https://github.com/AscensionGameDev/Intersect-Engine/blob/f9349c8d949c34c2586a621d42006153e35c9cfb/Intersect.Client/Entities/Entity.cs#L1423-L1437

But ShouldNotDrawHpBar is only true (resulting in a short-circuit to ShouldDrawHpBar to false) if HP is 0, HP is max, or if there is a shield -- which is actually where the problem lies.

![image](https://user-images.githubusercontent.com/1476550/210444921-f2e41bd0-acdf-4459-8e3b-0e7df68b656a.png)

I did not catch this in my review, but you maintained `|| shield > 0` when it needed to be `(<other conditions>) && shield < 1` in ShouldNotDrawHpBar.

Reply:
it should always been shield < 1 back then when the method was named "should draw"
the hp bar isn't supposed to be drawn if the shield bar is up ('ve never personally seen it in action but from what i understand this is how the logic flows). thus, shield > 0 is what we need in order to tell that the HP bar should not be drawn since the shield bar is up instead.

Reply:
just doubled checked all cases, not sure what i was into that day, couldn't really repro the issues i was experiencing today -facepalm- so i'm closing this one.